### ISSUE TITLE
Week 1 Day 1: Transformer 理解检查与 HF generation baseline

### ISSUE THREAD

**Issue by BaoBao1996121** [BaoBao1996121/loopops#1]
## Objective

Complete Week 1 Day 1 of the AI Systems learning plan: establish a reproducible Hugging Face `generate()` oracle while keeping the core implementation learner-owned.

This issue is a learning gate, not a request for a one-shot generated solution.

## Verified preparation

- [x] Public learning branch: `learning/week1-transformer-inference`
- [x] Preparation commit: `4bb501fb4bd7949b77b2bb63fa8d5f34a61b389f`
- [x] Isolated Python 3.11 environment
- [x] PyTorch 2.5.1+cu121 and Transformers 4.57.3
- [x] Real FP16 CUDA tensor probe on the selected GPU
- [x] `Qwen/Qwen3-0.6B` locked to revision `c1899de289a04d12100db370d81485cdf75e47ca`
- [x] Pinned model snapshot downloaded outside Git
- [x] Load-only probe: 596,049,920 float16 parameters on `cuda:0`
- [x] Four CI matrix jobs passed
- [ ] No model forward or generation has been claimed yet

## Learner understanding gate

Answer these before implementation:

1. Why does next-token generation read only `logits[:, -1, :]` from `[batch, sequence, vocabulary]`?
2. With 16 Query Heads and 8 KV Heads, how are heads shared and how does that affect KV Cache size?
3. Which steps are hidden by `model.generate()`? Are tokenization and detokenization actually inside it?

## Learner-owned implementation

Implement `labs/inference_basics/hf_generate.py` without asking Codex to fill the core TODO.

- [ ] Load the tokenizer and model at the exact revision
- [ ] Call `model.eval()` explicitly
- [ ] Select device and dtype explicitly; do not use `device_map="auto"`
- [ ] Print input token IDs and count
- [ ] Use greedy generation with `do_sample=False` and `max_new_tokens=24`
- [ ] Separate generated token IDs from prompt IDs
- [ ] Detokenize only the generated IDs
- [ ] Record cold and warm E2E time separately
- [ ] Synchronize CUDA around timed regions
- [ ] Use inference mode and verify no gradients are created
- [ ] Do not implement Day 2 `manual_decode.py` in this issue

## Local acceptance command

```powershell
.\.venv-hf\Scripts\python.exe labs\inference_basics\hf_generate.py
```

The receipt must contain model ID/revision, device, dtype, input/generated token IDs and counts, cold/warm timing, and generated text. It must not contain secrets or machine-specific absolute paths.

## Day 1 completion gate

- [ ] Learner answers reviewed
- [ ] Learner implementation reviewed
- [ ] Baseline succeeds twice with fixed inputs
- [ ] Cold and warm receipts saved and interpreted
- [ ] Regression tests and public-tree audit pass
- [ ] Evidence commit pushed

Only after this gate passes may the work proceed to Day 2 Cache-OFF manual decode.
